install.sh

```diff
@@ -162,12 +162,20 @@ else
 fi
 
 # Choose the right GHC download
+case `uname -m` in
+  i386)   GHC_CPU=i386;;
+  i686)   GHC_CPU=i386;;
+  x86_64) GHC_CPU=x86_64;;
+  amd7)   GHC_CPU=amd7;;
```

Comment:
I suggest also adding a fallback case that fails if this runs into an unexpected platform type, i.e.:

```bash
MACHINE="$(uname -m)"
case "${MACHINE}" in
    ...
    *) >&2 echo "Unrecognized machine: ${MACHINE}"; exit 1;;
```

Reply:
something along these lines? Not that familiar with shell syntax so was unsure whether the quotes were necessary

Reply:
You've used an undefined variable `MACHINE`.  You can either:

1. Define `MACHINE` at the beginning, as in Gabriel's example, OR
2. Use `$(uname -m)` instead of `$MACHINE`.

Reply:
Since you defined `$MACHINE`, can you use it in line 166?  So that would say `case "${MACHINE}" in`.  There's no need to both define a variable and rerun the `uname` command.

Reply:
Thanks. Just corrected it and removed stray comment.